### PR TITLE
perl: fix mac patch

### LINKED ISSUE #30021: Installation issue: perl fails to build on M1 Pro
### Steps to reproduce the issue

```console
$ spack spec -l dealii
Input spec
--------------------------------
dealii

Concretized
--------------------------------
6u6y6z2  dealii@9.3.3%apple-clang@13.1.6+adol-c+arborx+arpack+assimp~cuda~doc+examples+ginkgo~gmsh+gsl+hdf5~int64~ipo+metis+mpi+muparser~nanoflann~netcdf+oce~optflags+p4est+petsc~python+scalapack+simplex+slepc+sundials+symengine+threads+trilinos build_type=DebugRelease cxxstd=default arch=darwin-monterey-m1
zuiqsso      ^adol-c@2.7.2%apple-clang@13.1.6~advanced_branching+atrig_erf~boost+doc+examples~openmp~sparse+stdczero arch=darwin-monterey-m1
2ouzct2          ^gnuconfig@2021-08-14%apple-clang@13.1.6 arch=darwin-monterey-m1
hoxkahs      ^arborx@1.1%apple-clang@13.1.6~cuda~ipo+mpi~openmp~rocm+serial~sycl+trilinos build_type=RelWithDebInfo arch=darwin-monterey-m1
zooyxvb          ^cmake@3.23.0%apple-clang@13.1.6~doc+ncurses+ownlibs~qt build_type=Release arch=darwin-monterey-m1
magn2um              ^ncurses@6.2%apple-clang@13.1.6~symlinks+termlib abi=none arch=darwin-monterey-m1
ecbz6ae                  ^pkgconf@1.8.0%apple-clang@13.1.6 arch=darwin-monterey-m1
nrwyzxr              ^openssl@1.1.1n%apple-clang@13.1.6~docs~shared certs=system arch=darwin-monterey-m1
4sj2mwy                  ^perl@5.34.1%apple-clang@13.1.6+cpanm+shared+threads arch=darwin-monterey-m1
wzqpihx                      ^berkeley-db@18.1.40%apple-clang@13.1.6+cxx~docs+stl patches=b231fcc arch=darwin-monterey-m1
ezmkkpu                      ^bzip2@1.0.8%apple-clang@13.1.6~debug~pic+shared arch=darwin-monterey-m1
ngafgpe                          ^diffutils@3.8%apple-clang@13.1.6 arch=darwin-monterey-m1
76t2ha6                              ^libiconv@1.16%apple-clang@13.1.6 libs=shared,static arch=darwin-monterey-m1
tdcubr7                      ^gdbm@1.19%apple-clang@13.1.6 arch=darwin-monterey-m1
76lorqz                          ^readline@8.1%apple-clang@13.1.6 arch=darwin-monterey-m1
ngw5pv4                      ^zlib@1.2.12%apple-clang@13.1.6+optimize+pic+shared patches=0d38234 arch=darwin-monterey-m1
kpwbl45          ^openmpi@4.1.3%apple-clang@13.1.6~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~java~legacylaunchers~lustre~memchecker~pmi~pmix+romio~singularity~sqlite3+static~thread_multiple+vt+wrapper-rpath fabrics=none schedulers=none arch=darwin-monterey-m1
thkd4wf              ^hwloc@2.7.1%apple-clang@13.1.6~cairo~cuda~gl~libudev+libxml2~netloc~nvml~opencl~pci~rocm+shared arch=darwin-monterey-m1
4367kxw                  ^libxml2@2.9.12%apple-clang@13.1.6~python arch=darwin-monterey-m1
5seelmz                      ^xz@5.2.5%apple-clang@13.1.6~pic libs=shared,static arch=darwin-monterey-m1
rq7lgut              ^libevent@2.1.12%apple-clang@13.1.6+openssl arch=darwin-monterey-m1
opsa7bv              ^openssh@8.9p1%apple-clang@13.1.6 arch=darwin-monterey-m1
vjrwmdw                  ^libedit@3.1-20210216%apple-clang@13.1.6 arch=darwin-monterey-m1
drgjher          ^trilinos@13.0.1%apple-clang@13.1.6~adios2+amesos+amesos2+anasazi+aztec~basker+belos~boost~chaco~complex~cuda~cuda_rdc~debug~dtk+epetra+epetraext~epetraextbtf~epetraextexperimental~epetraextgraphreorderings~exodus+explicit_template_instantiation~float+fortran~gtest~hdf5~hypre+ifpack+ifpack2~intrepid~intrepid2~ipo~isorropia+kokkos+mesquite~minitensor+ml+mpi+muelu~mumps~nox~openmp~panzer~phalanx~piro~python~rocm~rocm_rdc+rol~rythmos+sacado~scorec~shards+shared~shylu~stk~stokhos~stratimikos~strumpack~suite-sparse~superlu~superlu-dist~teko~tempus~thyra+tpetra~trilinoscouplings~wrapper~x11~zoltan~zoltan2 build_type=RelWithDebInfo cxxstd=14 gotype=long_long arch=darwin-monterey-m1
yzyuuhg              ^openblas@0.3.20%apple-clang@13.1.6~bignuma~consistent_fpcsr~ilp64+locking+pic+shared symbol_suffix=none threads=none arch=darwin-monterey-m1
7hwf372      ^arpack-ng@3.8.0%apple-clang@13.1.6+mpi+shared arch=darwin-monterey-m1
qitvngi      ^assimp@5.2.2%apple-clang@13.1.6~ipo+shared build_type=RelWithDebInfo patches=a227714 arch=darwin-monterey-m1
nc4fwvk          ^boost@1.76.0%apple-clang@13.1.6+atomic+chrono~clanglibcpp~container~context~coroutine+date_time~debug+exception~fiber+filesystem+graph~icu+iostreams+locale+log+math~mpi+multithreaded~numpy~pic+program_options~python+random+regex+serialization+shared+signals~singlethreaded+system~taggedlayout+test+thread+timer~versionedlayout+wave cxxstd=98 patches=a440f96 visibility=hidden arch=darwin-monterey-m1
7wgutof      ^ginkgo@1.4.0%apple-clang@13.1.6~cuda~develtools~full_optimizations~hwloc~ipo~oneapi~openmp~rocm+shared build_type=Release patches=43e571a arch=darwin-monterey-m1
bhvkke5      ^gsl@2.7%apple-clang@13.1.6~external-cblas arch=darwin-monterey-m1
cev6ojz      ^hdf5@1.12.1%apple-clang@13.1.6+cxx+fortran+hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_type=RelWithDebInfo patches=ee351eb arch=darwin-monterey-m1
sy2yifu      ^intel-tbb@2020.3%apple-clang@13.1.6~ipo+shared+tm build_type=RelWithDebInfo cxxstd=default patches=62ba015,ce1fb16,d62cb66 arch=darwin-monterey-m1
rzqvspc      ^metis@5.1.0%apple-clang@13.1.6~gdb~int64~real64+shared build_type=Release patches=4991da9 arch=darwin-monterey-m1
dqc3t7c      ^muparser@2.2.6.1%apple-clang@13.1.6 arch=darwin-monterey-m1
ydvqphc      ^netlib-scalapack@2.2.0%apple-clang@13.1.6~ipo~pic+shared build_type=Release patches=072b006,1c9ce5f arch=darwin-monterey-m1
lagtjrl      ^ninja@1.10.2%apple-clang@13.1.6 arch=darwin-monterey-m1
qvdxsqe          ^python@3.9.12%apple-clang@13.1.6+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=darwin-monterey-m1
cc6qfkl              ^apple-libuuid@1353.100.2%apple-clang@13.1.6 arch=darwin-monterey-m1
4rm6kzf              ^expat@2.4.8%apple-clang@13.1.6~libbsd arch=darwin-monterey-m1
xgi2agw              ^gettext@0.21%apple-clang@13.1.6+bzip2+curses+git~libunistring+libxml2+tar+xz arch=darwin-monterey-m1
77cdgz2                  ^tar@1.34%apple-clang@13.1.6 arch=darwin-monterey-m1
hvncl5m              ^libffi@3.4.2%apple-clang@13.1.6 arch=darwin-monterey-m1
ennkfzs              ^sqlite@3.37.2%apple-clang@13.1.6+column_metadata+dynamic_extensions+fts~functions+rtree arch=darwin-monterey-m1
3vslcez      ^oce@0.18.3%apple-clang@13.1.6~X11+tbb arch=darwin-monterey-m1
abakina      ^p4est@2.8%apple-clang@13.1.6+mpi~openmp arch=darwin-monterey-m1
vl5y4ck          ^libtool@2.4.7%apple-clang@13.1.6 arch=darwin-monterey-m1
e4qytr2              ^m4@1.4.19%apple-clang@13.1.6+sigsegv patches=9dc5fbd,bfdffa7 arch=darwin-monterey-m1
mdobjrv                  ^libsigsegv@2.13%apple-clang@13.1.6 arch=darwin-monterey-m1
siozfc5      ^petsc@3.17.0%apple-clang@13.1.6~X~batch~cgns~complex~cuda~debug+double~exodusii~fftw+fortran~giflib+hdf5~hpddm~hwloc+hypre~int64~jpeg~knl~kokkos~libpng~libyaml~memkind+metis~mkl-pardiso~mmg~moab~mpfr+mpi+mumps~openmp+p4est~parmmg~ptscotch~random123~rocm~saws~scalapack+shared~strumpack+suite-sparse+superlu-dist~tetgen~trilinos~valgrind clanguage=C arch=darwin-monterey-m1
3owltmz          ^hypre@2.24.0%apple-clang@13.1.6~complex~cuda~debug+fortran~gptune~int64~internal-superlu~mixedint+mpi~openmp~rocm~shared~superlu-dist~unified-memory arch=darwin-monterey-m1
7yugfv7          ^mumps@5.4.1%apple-clang@13.1.6~blr_mt+complex+double+float~int64+metis+mpi~openmp+parmetis~ptscotch~scotch+shared patches=1946864 arch=darwin-monterey-m1
yztnzxz              ^parmetis@4.0.3%apple-clang@13.1.6~gdb~int64~ipo+shared build_type=RelWithDebInfo patches=4f89253,50ed208,704b84f arch=darwin-monterey-m1
j3hhbur          ^suite-sparse@5.10.1%apple-clang@13.1.6~cuda~graphblas~openmp+pic~tbb arch=darwin-monterey-m1
bdy3b6s              ^gmp@6.2.1%apple-clang@13.1.6+shared~static arch=darwin-monterey-m1
uoucic5                  ^autoconf@2.69%apple-clang@13.1.6 patches=35c4492,7793209,a49dd5b arch=darwin-monterey-m1
tvv2qcm                  ^automake@1.16.5%apple-clang@13.1.6 arch=darwin-monterey-m1
skrla3j              ^mpfr@4.1.0%apple-clang@13.1.6+shared~static arch=darwin-monterey-m1
kpbbgd5                  ^autoconf-archive@2019.01.06%apple-clang@13.1.6 arch=darwin-monterey-m1
ppunc33                  ^texinfo@6.5%apple-clang@13.1.6 patches=12f6edb,1732115 arch=darwin-monterey-m1
hvsx3me          ^superlu-dist@7.2.0%apple-clang@13.1.6~cuda~int64~ipo~openmp~rocm+shared build_type=RelWithDebInfo patches=8da9e21 arch=darwin-monterey-m1
duyfcyf      ^slepc@3.17.0%apple-clang@13.1.6+arpack~blopex~cuda~rocm arch=darwin-monterey-m1
d4h3kjz      ^sundials@5.8.0%apple-clang@13.1.6+ARKODE+CVODE+CVODES+IDA+IDAS+KINSOL~cuda+examples+examples-install~f2003~fcmix+generic-math~hypre~int64~ipo~klu~lapack~magma~monitoring+mpi~openmp~petsc~pthread~raja~rocm+shared+static~superlu-dist~superlu-mt~sycl~trilinos build_type=RelWithDebInfo cstd=99 cxxstd=14 precision=double arch=darwin-monterey-m1
7fhn5px      ^symengine@0.8.1%apple-clang@13.1.6~boostmp~flint~ipo~llvm+mpc+mpfr~openmp~piranha+shared+thread_safe build_type=Release arch=darwin-monterey-m1
lqvnfv2          ^mpc@1.2.1%apple-clang@13.1.6 arch=darwin-monterey-m1
```
The command I ran
```
spack install --only dependencies -j8 --fail-fast dealii
```

### Error message

<details><summary>Error message</summary><pre>
==> Installing perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph
==> No binary for perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph found: installing from source
==> Using cached archive: /Users/praveen/Applications/spack/var/spack/cache/_source-cache/archive/35/357951a491b0ba1ce3611263922feec78ccd581dddc24a446b033e25acf242a1.tar.gz
==> Using cached archive: /Users/praveen/Applications/spack/var/spack/cache/_source-cache/archive/9d/9da50e155df72bce55cb69f51f1dbb4b62d23740fb99f6178bb27f22ebdf8a46.tar.gz
==> Moving resource stage
	source: /var/folders/hh/bdt5s8c5131gxplnt9jm24580000gn/T/praveen/spack-stage/resource-cpanm-4sj2mwyciz7cjystlzor6wkoguiv2vph/spack-src/
	destination: /var/folders/hh/bdt5s8c5131gxplnt9jm24580000gn/T/praveen/spack-stage/spack-stage-perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph/spack-src/cpanm/cpanm
==> Ran patch() for perl
==> perl: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    './Configure' '-des' '-Dprefix=/Users/praveen/Applications/spack/opt/spack/darwin-monterey-m1/apple-clang-13.1.6/perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph' '-Dlocincpth=/Users/praveen/Applications/spack/opt/spack/darwin-monterey-m1/apple-clang-13.1.6/gdbm-1.19-tdcubr74rbxd5veryp7zw6qijtm2gvfx/include' '-Dloclibpth=/Users/praveen/Applications/spack/opt/spack/darwin-monterey-m1/apple-clang-13.1.6/gdbm-1.19-tdcubr74rbxd5veryp7zw6qijtm2gvfx/lib' '-Accflags=-DAPPLLIB_EXP=\"/Users/praveen/Applications/spack/opt/spack/darwin-monterey-m1/apple-clang-13.1.6/perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph/lib/perl5\"' '-Duseshrplib' '-Dusethreads'
See build log for details:
  /var/folders/hh/bdt5s8c5131gxplnt9jm24580000gn/T/praveen/spack-stage/spack-stage-perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph/spack-build-out.txt

==> Warning: Skipping build of openblas-0.3.20-yzyuuhgr4op4ju5qm7gsnfssqyxkkv3d since perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph failed
==> Warning: Skipping build of mumps-5.4.1-7yugfv7jczz6m7uq3irlfcpg3xbs3pzy since openblas-0.3.20-yzyuuhgr4op4ju5qm7gsnfssqyxkkv3d failed
==> Warning: Skipping build of petsc-3.17.0-siozfc5hgg3s5hpx6xkq4dr63wvlpox5 since mumps-5.4.1-7yugfv7jczz6m7uq3irlfcpg3xbs3pzy failed
==> Warning: Skipping build of slepc-3.17.0-duyfcyfovdykivjrskriqbnkjfzv5hk2 since petsc-3.17.0-siozfc5hgg3s5hpx6xkq4dr63wvlpox5 failed
==> Warning: Skipping build of trilinos-13.0.1-drgjhertuo2ilv4zmvwz22jynedzfleb since openblas-0.3.20-yzyuuhgr4op4ju5qm7gsnfssqyxkkv3d failed
==> Warning: Skipping build of arborx-1.1-hoxkahslcdcenbo2smfhhder2sed27qs since trilinos-13.0.1-drgjhertuo2ilv4zmvwz22jynedzfleb failed
==> Warning: Skipping build of arpack-ng-3.8.0-7hwf372n3tf36lqcmmesj3hgo4enliip since openblas-0.3.20-yzyuuhgr4op4ju5qm7gsnfssqyxkkv3d failed
==> Warning: Skipping build of superlu-dist-7.2.0-hvsx3megnx36znb6lfobkrx7zn4gsfxx since openblas-0.3.20-yzyuuhgr4op4ju5qm7gsnfssqyxkkv3d failed
==> Warning: Skipping build of netlib-scalapack-2.2.0-ydvqphcufmkv72rjjork6iwsz6iylbnw since openblas-0.3.20-yzyuuhgr4op4ju5qm7gsnfssqyxkkv3d failed
==> Warning: Skipping build of hypre-2.24.0-3owltmzf2pbjt73fl7fc3x6nomzq3jos since openblas-0.3.20-yzyuuhgr4op4ju5qm7gsnfssqyxkkv3d failed
==> Warning: Skipping build of suite-sparse-5.10.1-j3hhburpcsaxs2u7fj2uo6ft2o2i4so4 since openblas-0.3.20-yzyuuhgr4op4ju5qm7gsnfssqyxkkv3d failed
==> Warning: Skipping build of texinfo-6.5-ppunc334rbc7pya3cguy4vwo3fee2ll3 since perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph failed
==> Warning: Skipping build of mpfr-4.1.0-skrla3jxund5kllcyvshhaso6fnvepfc since texinfo-6.5-ppunc334rbc7pya3cguy4vwo3fee2ll3 failed
==> Warning: Skipping build of symengine-0.8.1-7fhn5px2wjgvkjqsytjd7q6hqklilttg since mpfr-4.1.0-skrla3jxund5kllcyvshhaso6fnvepfc failed
==> Warning: Skipping build of mpc-1.2.1-lqvnfv2vfj6e6emk7gqkw7vdd6zp6dks since mpfr-4.1.0-skrla3jxund5kllcyvshhaso6fnvepfc failed
==> Warning: Skipping build of openssl-1.1.1n-nrwyzxrk2tccunksac4vd642o6es3ncj since perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph failed
==> Warning: Skipping build of openssh-8.9p1-opsa7bv5lydquwgd4qxaxt746ulo45ys since openssl-1.1.1n-nrwyzxrk2tccunksac4vd642o6es3ncj failed
==> Warning: Skipping build of openmpi-4.1.3-kpwbl45k4qxve3yqo5anpxsqavvwhmi4 since openssh-8.9p1-opsa7bv5lydquwgd4qxaxt746ulo45ys failed
==> Warning: Skipping build of parmetis-4.0.3-yztnzxzo7cd7rimwpcokl7ndyj7vscj4 since openmpi-4.1.3-kpwbl45k4qxve3yqo5anpxsqavvwhmi4 failed
==> Warning: Skipping build of sundials-5.8.0-d4h3kjzvy42s4hjf57gzras65tu2awsj since openmpi-4.1.3-kpwbl45k4qxve3yqo5anpxsqavvwhmi4 failed
==> Warning: Skipping build of hdf5-1.12.1-cev6ojzrdulyonbz2h25izavqylfz4xi since openmpi-4.1.3-kpwbl45k4qxve3yqo5anpxsqavvwhmi4 failed
==> Warning: Skipping build of p4est-2.8-abakinafobcdqqm56qqv3syc5imacdui since openmpi-4.1.3-kpwbl45k4qxve3yqo5anpxsqavvwhmi4 failed
==> Warning: Skipping build of libevent-2.1.12-rq7lgututf3tovyp22mcock5aygy6fz4 since openssl-1.1.1n-nrwyzxrk2tccunksac4vd642o6es3ncj failed
==> Warning: Skipping build of cmake-3.23.0-zooyxvbojh4fyy4fg3ijss53chz73fqg since openssl-1.1.1n-nrwyzxrk2tccunksac4vd642o6es3ncj failed
==> Warning: Skipping build of oce-0.18.3-3vslcezn2oeedxxfeobo6fqxpiz5nikg since cmake-3.23.0-zooyxvbojh4fyy4fg3ijss53chz73fqg failed
==> Warning: Skipping build of metis-5.1.0-rzqvspcvl3astlhf4bhsbjvihylbwzc4 since cmake-3.23.0-zooyxvbojh4fyy4fg3ijss53chz73fqg failed
==> Warning: Skipping build of assimp-5.2.2-qitvngie7bx753tskbfena6adnak3btj since cmake-3.23.0-zooyxvbojh4fyy4fg3ijss53chz73fqg failed
==> Warning: Skipping build of muparser-2.2.6.1-dqc3t7coirejrde6otedluspodfrtpvg since cmake-3.23.0-zooyxvbojh4fyy4fg3ijss53chz73fqg failed
==> Warning: Skipping build of intel-tbb-2020.3-sy2yifu2jtxxmk476cfpxyx6w3a5i33p since cmake-3.23.0-zooyxvbojh4fyy4fg3ijss53chz73fqg failed
==> Warning: Skipping build of ginkgo-1.4.0-7wgutofkf2xdx6ftokbt6uwu4tcz7ycg since cmake-3.23.0-zooyxvbojh4fyy4fg3ijss53chz73fqg failed
==> Warning: Skipping build of python-3.9.12-qvdxsqes3gbf5gf2iit6r6kiw24qqmfr since openssl-1.1.1n-nrwyzxrk2tccunksac4vd642o6es3ncj failed
==> Warning: Skipping build of ninja-1.10.2-lagtjrldxs6rk6vjmomrysz4vfkaz3ni since python-3.9.12-qvdxsqes3gbf5gf2iit6r6kiw24qqmfr failed
==> Warning: Skipping build of automake-1.16.5-tvv2qcmbyoocu5pngfxpb4qftnogsmln since perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph failed
==> Warning: Skipping build of gmp-6.2.1-bdy3b6sl24e5vxu4qtvontez77oyjnac since automake-1.16.5-tvv2qcmbyoocu5pngfxpb4qftnogsmln failed
==> Warning: Skipping build of autoconf-2.69-uoucic5imcypt2zp57ev7br22ijygwsp since perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph failed
==> Error: Terminating after first install failure: ProcessError: Command exited with status 1:
    './Configure' '-des' '-Dprefix=/Users/praveen/Applications/spack/opt/spack/darwin-monterey-m1/apple-clang-13.1.6/perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph' '-Dlocincpth=/Users/praveen/Applications/spack/opt/spack/darwin-monterey-m1/apple-clang-13.1.6/gdbm-1.19-tdcubr74rbxd5veryp7zw6qijtm2gvfx/include' '-Dloclibpth=/Users/praveen/Applications/spack/opt/spack/darwin-monterey-m1/apple-clang-13.1.6/gdbm-1.19-tdcubr74rbxd5veryp7zw6qijtm2gvfx/lib' '-Accflags=-DAPPLLIB_EXP=\"/Users/praveen/Applications/spack/opt/spack/darwin-monterey-m1/apple-clang-13.1.6/perl-5.34.1-4sj2mwyciz7cjystlzor6wkoguiv2vph/lib/perl5\"' '-Duseshrplib' '-Dusethreads'
</pre></details>

### Information on your system

```
* **Spack:** 0.18.0.dev0 (7893be7712ed709f6136ac83f49afc3d718d5ddc)
* **Python:** 3.9.10
* **Platform:** darwin-monterey-m1
* **Concretizer:** clingo
```

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/8469065/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/8469067/spack-build-env.txt)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by sethrj**
- Apply macosx version workaround to all 5.34 and less
 - Delete an obsolete mac patch file that I didn't notice when adding #28797

Closes #29887

### CODE REVIEW

**Comment by albestro** [comment]
Please, have a look at https://github.com/spack/spack/issues/30021#issuecomment-1096955786.

File: var/spack/repos/builtin/packages/perl/package.py [hunk 1]
```diff
@@ -372,8 +368,7 @@ def setup_build_environment(self, env):
 
         spec = self.spec
 
-        if (spec.version <= Version('5.34.0')
-            and spec.platform == 'darwin'
+        if (spec.satisfies('@:5.34 platform=darwin')
```
Comment (albestro) [inline var/spack/repos/builtin/packages/perl/package.py:371]: I think you correctly "ported" the code as it was before. Anyway, apparently even on newer versions of perl this change is required.

I would take the chance to improve this condition in this PR. I will try to investigate a bit to understand what are the perl versions requiring this "patch".

Comment (sethrj) [reply var/spack/repos/builtin/packages/perl/package.py:371]: The `@:5.34` includes everything through `5.34.99999` (i.e. the failing perl implementation) so it's a superset of your patch. It *probably* won't hurt to make this change to `MACOSX_DEPLOYMENT_TARGET` for newer versions of Perl, but I've been hoping the developers would fix it sooner.

**Comment by sethrj** [comment]
@adamjstewart Would you be able to test this and merge? More people [have encountered](https://github.com/spack/spack/issues/30021) the issue this solves since I submitted the PR.

**Comment by adamjstewart** [comment]
I don't have macOS 11 so I can't really test this properly. Maybe @ax3l can test?

**Comment by sethrj** [comment]
I tested it out on 12.2.1 and it worked for me, so I'm happy for anyone to rubberstamp this so we don't continue to accumulate perl build errors 😬

**Comment by alalazo** [comment]
> Only mark most recent version as preferred

That was done on purpose. I think for `perl` even versions are considered "stable" and odd "develop".

**Comment by sethrj** [comment]
> > Only mark most recent version as preferred
> 
> That was done on purpose. I think for `perl` even versions are considered "stable" and odd "develop".

Right, I meant the most recent *even* version. A bunch of older stable versions were also marked as "preferred" so there were 20 preferred versions.

**Comment by adamjstewart** [comment]
But those older even versions should be preferred over older odd versions if someone asks for a specific range of Perl versions.

**Comment by sethrj** [comment]
Aha, I didn't realize the version resolution worked like that. I'll revert the change. Thanks for the heads up!

File: var/spack/repos/builtin/packages/perl/package.py (hunk 1 above)
Comment (albestro) [reply var/spack/repos/builtin/packages/perl/package.py:371]: Ah you're right, I always confuse

```python
>>> Version("3.4.1") <= Version("3.4")
False
>>> Version("3.4.1").satisfies(Version("3.4"))
True
```

Thanks!

**Review by alalazo — APPROVED** [review]

**Comment by sethrj** [comment]
@spackbot re-run pipeline

**Review by albestro — APPROVED** [review]

**Pull request merged**